cmd/cli/uninstall_mesh.go

```diff
@@ -118,13 +118,11 @@ func (d *uninstallMeshCmd) run() error {
 			fmt.Fprintf(d.out, "No OSM control planes found\n")
 			return nil
 		}
+
 		// Searches for the mesh specified by the mesh-name flag if specified
-		specifiedMeshFound := false
-		if d.meshName != "" {
-			specifiedMeshFound = d.findMesh(meshInfoList)
-			if !specifiedMeshFound {
-				return nil
-			}
+		specifiedMeshFound := d.findSpecifiedMesh(meshInfoList)
+		if !specifiedMeshFound {
+			return nil
```

Comment:
I am not sure this change has the same functionality as before. Previously, if a mesh name was specified, we'd search for the mesh and verify its existence. If it didn't exist we would return nil and not perform an uninstall. With this change, we call `findSpecifiedMesh` regardless of whether or not a mesh name was specified. If a mesh name was not specified `findSpecifiedMesh` will return false and the following condition check with be met and we will return without uninstalling. I think this may be the cause of the failing tests in `e2e_osm_reinstall_test.go`.